spec/tests/helpers/perf_run_helper_spec.rb

```diff
@@ -438,33 +438,76 @@ class PerfRunHelperClass
+        expect(FileUtils).to receive(:mkdir_p)
+        subject.copy_system_logs(host)
+      end
+    end
+
+    context "when archiving fails" do
+      before { allow(subject).to receive(:archive_system_logs).with(host).and_return(nil) }
+
+      it "does not call scp_from" do
+        expect(subject).not_to receive(:scp_from)
+        subject.copy_system_logs(host)
+      end
+
+      it "does not call mkdir_p" do
+        expect(FileUtils).not_to receive(:mkdir_p)
+        subject.copy_system_logs(host)
+      end
+    end
+  end
+
+  describe "#archive_system_logs" do
+    let(:host) { double.as_null_object }
+    let(:puppet_logdir) { "/var/log/puppetlabs/puppet" }
+    let(:archive_dir) { "/var/log" }
+    let(:archive_name) { "puppet_logdir.tgz" }
+    let(:archive_path) { File.join(archive_dir, archive_name) }
+
+    context "when archiving succeeds" do
+      it "returns archive path" do
+        expect(subject).to receive(:archive_system_logs).with(host).and_return(archive_path)
+        subject.archive_system_logs(host)
+      end
+      it "runs tar on the host" do
+        result = Beaker::Result.new(host, "foo")
+        result.stdout = puppet_logdir
+        cmd1 = "rm -f #{archive_path}"
+        cmd2 = "cd #{archive_dir} && tar -czf #{archive_name} puppetlabs"
+        allow(subject).to receive(:puppet).and_return(puppet_logdir)
+        expect(subject).to receive(:on).with(host, puppet_logdir).ordered.and_return(result)
+        expect(subject).to receive(:on).with(host, cmd1, accept_all_exit_codes: true).ordered
+        expect(subject).to receive(:on).with(host, cmd2, accept_all_exit_codes: true).ordered
+        subject.archive_system_logs(host)
+      end
     end
-    it "calls archive_file_from with given host and parent dir of puppet_logdir" do
-      # test variables
-      build_id = gplt_timestamp = nil
-      puppet_logdir = "/foo/bar/baz"
-      archive_root = "/test/archive"
-      job_name = "foobar"
-      archive_name = "#{job_name}__#{build_id}__#{gplt_timestamp}__system_logs.tgz"
-      ENV["JOB_NAME"] = job_name
-      myhost = hosts.first
-      # mock out @archive_root instance variable
-      subject.instance_variable_set(:@archive_root, archive_root)
-      # mock out beaker result
-      result = Beaker::Result.new(myhost, "foo")
-      result.stdout = puppet_logdir
-
-      allow(subject).to receive(:on).with(myhost, any_args).and_return(result)
-      expect(subject).to receive(:puppet).with("config", "print", "logdir")
-      expect(subject).to receive(:archive_file_from).with(myhost,
-                                                          File.dirname(puppet_logdir),
-                                                          {}, archive_root, archive_name)
-      subject.copy_system_logs(myhost)
+    context "when archiving fails" do
+      it "returns nil" do
+        expect(subject).to receive(:archive_system_logs).with(host).and_return(nil)
+        subject.archive_system_logs(host)
```

Comment:
Same as above.  I think this mocks out the method under test.
I think this should be
expect(subject.archive_system_logs(host)).to eq(nil)  
or something like it.  nil might be a special case...

Reply:
Indeed.  Update pushed to resolve this.